shared/actions/signup/index.js

```diff
@@ -242,8 +242,12 @@ function signup (skipMail): TypedAsyncAction<Signup | ShowPaperKey> {
   })
 }
 
-export function resetSignup (): ResetSignup {
-  return {type: Constants.resetSignup, payload: {}}
+export function resetSignup (): TypedAsyncAction<ResetSignup | RouteAppend> {
+  return dispatch => new Promise((resolve, reject) => {
+    dispatch({type: Constants.resetSignup, payload: {}})
+    dispatch(nextPhase())
```

Comment:
The first line is us changing the phase. that is changing the routing. The index should just listen to phase changing and change the component it shows instead of using the route, or the action should change the route and not set phase, we shouldn't do both